src/platform.ts

```diff
@@ -81,13 +82,14 @@ export class MideaPlatform implements DynamicPlatformPlugin {
     this.log.info('Start device discovery...');
     // If IP address is in config then probe them directly
     this.platformConfig.devices.forEach((device) => {
+      device = defaultsDeep(device, defaultDeviceConfig);
```

Comment:
Why bother?

Reply:
The question mark handling of advanced_options and ip isn't very describing because in the DeviceConfig type those fields are not optional.